endpoint/domain_filter.go

```diff
@@ -199,3 +199,24 @@ func (df *DomainFilter) UnmarshalJSON(b []byte) error {
 	*df = NewRegexDomainFilter(include, exclude)
 	return nil
 }
+
+func (df DomainFilter) MatchParent(domain string) bool {
+	if matchFilter(df.exclude, domain, false) {
+		return false
+	}
+	if len(df.Filters) == 0 {
+		return true
+	}
+
+	strippedDomain := strings.ToLower(strings.TrimSuffix(domain, "."))
+	for _, filter := range df.Filters {
+		if filter == "" || strings.HasPrefix(filter, ".") {
+			// We don't check parents if the filter is prefixed with "."
```

Comment:
Why we do not check it would be the important comment. Right now I see this in the code, but now why :)

Reply:
When we put a dot "." as a prefix in the `--domain-filter=.x.example.com,xpto.example.com`, the MatchParent function ignores `x.example.com` domain as a filter but not `xpto.example.com`.